src/components/Buttons/SubmitButton.jsx

```diff
@@ -25,26 +43,40 @@ export const SubmitButton = ({ color = "primary", isLoading, className, children
             <span
                 className={clsx(
                     "absolute inset-0 flex items-center justify-center",
-                    isLoading ? "opacity-100" : "opacity-0",
+                    showTransition ? "opacity-100" : "opacity-0",
                 )}
             >
-                <Transition
-                    appear
-                    as="span"
-                    show={isLoading}
-                    enter="transition-all duration-700"
-                    enterFrom="opacity-0"
-                    enterTo="opacity-100"
-                >
-                    <Spinner size="current" color="current" className="relative -top-0.25 text-white" />
-                </Transition>
+                {showTransition && (
+                    <Transition
+                        appear
+                        as="span"
+                        show={showTransition}
+                        enter="transition-all duration-700"
+                        enterFrom="opacity-0"
+                        enterTo="opacity-100"
+                    >
+                        {showSuccess && (
+                            <CheckIcon
+                                size="medium"
+                                color="current"
+                                className={clsx("relative -top-0.25 text-white", {
+                                    "text-black": rest.variant === "outline",
```

Comment:
When you show the success icon, you are still able to click the button. This can cause multiple clicks and excessive API calls. Needs to be fixed. See my message in the slack channel